src/emqx_channel.erl

```diff
@@ -926,7 +926,13 @@ return_sub_unsub_ack(Packet, Channel) ->
        | {shutdown, Reason :: term(), Reply :: term(), emqx_types:packet(), channel()}).
 handle_call(kick, Channel) ->
     Channel1 = ensure_disconnected(kicked, Channel),
-    disconnect_and_shutdown(kicked, ok, Channel1);
+    case Channel1 of
```

Comment:
please add clause for `reason_code/1`, line:1674

Reply:
I.e `reason_code(kicked, ...` ? It seems a never matched clause if we add it?

Reply:
It means keep `disconnect_and_shutdown/3` here.
The fun will call `reason_code/1`.
So just add clause `reason_code(kicked) -> ?RC_ADMINISTRATIVE_ACTION;`